projects/packages/assets/src/class-assets.php

```diff
@@ -458,4 +485,196 @@ public static function wp_default_scripts_hook( $wp_scripts ) {
 		$wp_scripts->add_inline_script( 'wp-jp-i18n-state', $js, 'before' );
 	}
 
+	// endregion .
+
+	// ////////////////////
+	// region Textdomain aliasing
+
+	/**
+	 * Register a textdomain alias.
+	 *
+	 * Composer packages included in plugins will likely not use the textdomain of the plugin, while
+	 * WordPress's i18n infrastructure will include the translations in the plugin's domain. This
+	 * allows for mapping the package's domain to the plugin's.
+	 *
+	 * Since multiple plugins may use the same package, we include the package's version here so
+	 * as to choose the most recent translations (which are most likely to match the package
+	 * selected by jetpack-autoloader).
+	 *
+	 * @since $$next-version$$
+	 * @param string $from Domain to alias.
+	 * @param string $to Domain to alias it to.
+	 * @param string $totype What is the target of the alias: 'plugins', 'themes', or 'core'.
+	 * @param string $ver Version of the `$from` domain.
+	 * @throws InvalidArgumentException If arguments are invalid.
+	 */
+	public static function alias_textdomain( $from, $to, $totype, $ver ) {
+		if ( ! in_array( $totype, array( 'plugins', 'themes', 'core' ), true ) ) {
+			throw new InvalidArgumentException( 'Type must be "plugins", "themes", or "core"' );
+		}
+
+		if ( did_action( 'wp_default_scripts' ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				sprintf(
+					/* translators: 1: wp_default_scripts. 2: Name of the domain being aliased. */
+					esc_html__( 'Textdomain aliases should be registered before the %1$s hook. This notice was triggered by the %2$s domain.', 'jetpack' ),
+					'<code>wp_default_scripts</code>',
+					'<code>' . esc_html( $from ) . '</code>'
+				),
+				''
+			);
+		}
+
+		if ( empty( self::$domain_map[ $from ] ) ) {
+			self::init_domain_map_hooks( $from, array() === self::$domain_map );
+			self::$domain_map[ $from ] = array( $to, $totype, $ver );
+		} elseif ( Semver::compare( $ver, self::$domain_map[ $from ][2] ) > 0 ) {
```

Comment:
Do you think it would help to add an inline comment here explaining why we're going with that custom implementation?

Reply:
Custom implementation as in "why not use composer/semver", or custom implementaion as in "why not use PHP's broken version_compare()"?

Reply:
Answered both questions in 9e9c8170